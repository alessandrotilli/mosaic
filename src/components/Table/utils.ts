export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
