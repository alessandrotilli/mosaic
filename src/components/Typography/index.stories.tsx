import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import getDocsPage from "../../utils/stories";
import LocaleDecorator, { MessageMock } from "../../utils/stories/decorators/Locale";

import Typography, { DATA_CY_DEFAULT, LOCALIZABLE_PROPS } from ".";

const COMPONENT_NAME = "Typography";
Typography.displayName = COMPONENT_NAME;

const meta = {
  title: "Display/Typography",
  component: Typography,
  decorators: [LocaleDecorator],
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Typography Component",
          url: "https://mui.com/components/typography/",
        },
        component: COMPONENT_NAME,
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
        },
        localizableProps: LOCALIZABLE_PROPS,
      }),
      description: {
        component: "The Typography component is used to render text with different styles.",
      },
    },
  },
} satisfies Meta<typeof Typography>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    children: "Text example",
  },
};

export const BottomSpacing: Story = {
  args: {
    ...Primary.args,
    bottomSpacing: true,
  },
};

export const CustomContent: Story = {
  args: {
    ...Primary.args,
    children: (
      <span>
        <b>Strong</b> text
      </span>
    ),
  },
};

export const DisplayBlock: Story = {
  args: {
    ...Primary.args,
    display: "block",
  },
};

export const DisplayInline: Story = {
  args: {
    ...Primary.args,
    display: "inline",
  },
};

export const Loading: Story = {
  args: {
    ...Primary.args,
    loading: true,
  },
};

export const Localized: Story = {
  args: {
    ...Primary.args,
    localized: true,
    children: MessageMock.typography,
    content: MessageMock.typography,
  },
};

export const Truncated: Story = {
  args: {
    ...Primary.args,
    truncated: true,
  },
};

export const VariantBody: Story = {
  args: {
    ...Primary.args,
    variant: "body",
  },
};

export const VariantCaption: Story = {
  args: {
    ...Primary.args,
    variant: "caption",
  },
};

export const VariantLabel: Story = {
  args: {
    ...Primary.args,
    variant: "label",
  },
};

export const VariantOverline: Story = {
  args: {
    ...Primary.args,
    variant: "overline",
  },
};

export const VariantPagetitle: Story = {
  args: {
    ...Primary.args,
    variant: "pagetitle",
  },
};

export const VariantTitle: Story = {
  args: {
    ...Primary.args,
    variant: "title",
  },
};
