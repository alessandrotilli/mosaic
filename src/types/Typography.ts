import { ReactNode } from "react";

import { ILoadable, ILocalizable } from "./Base";

/**
 * @deprecated Use "body" | "caption" | "label" | "overline" | "pagetitle" | "title"
 */
export enum TypographyVariants {
  body = "body1",
  caption = "caption",
  label = "subtitle1",
  overline = "overline",
  pagetitle = "h5",
  title = "h6",
}

export type ITypographyVariants = "body" | "caption" | "label" | "overline" | "pagetitle" | "title";

/**
 * @deprecated Use "block" | "initial" | "inline"
 */
export enum TypographyDisplay {
  default = "initial",
  block = "block",
  inline = "inline",
}

type ITypographyDisplay = "block" | "initial" | "inline";

export interface ITypography extends ILoadable, ILocalizable {
  /**
   * Adds bottom spacing
   */
  bottomSpacing?: boolean;
  /**
   * Content of Typography, can also be expressed via children
   */
  content?: ReactNode;
  /**
   * Display mode
   */
  display?: ITypographyDisplay | TypographyDisplay;
  /**
   * Enables truncated mode (useful when container has width constraints)
   */
  truncated?: boolean;
  /**
   * Variant mode
   */
  variant?: ITypographyVariants | TypographyVariants;
}
