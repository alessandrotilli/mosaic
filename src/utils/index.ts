import { ChangeEvent, MouseEvent } from "react";

export interface ISubpart {
  label: string;
  value?: Function;
}

export interface ISubpartSuffix {
  label: string;
  suffix: string;
}

export type ISubpartMap<T extends string = string> = {
  [key in T]: ISubpart;
};

export const DATA_CY_SUFFIX_SEPARATOR = "-";

/**
 * Gets composed dataCy for components with subparts
 * @param dataCy the dataCy prefix
 * @param subpart the subpart to consider
 * @param args optional arguments for subpart evaluation
 */
export const getComposedDataCy = (dataCy: string, subpart: ISubpart, args?: any): string => {
  const { label, value } = subpart;
  return `${dataCy}${DATA_CY_SUFFIX_SEPARATOR}${!value ? slugify(label) : value(args)}`;
};

/**
 * Gets all composed dataCy for all listed subparts
 * @param subpartMap the map of all subparts to consider
 */
export const getAllComposedDataCy = (subpartMap: ISubpartMap): ISubpartSuffix[] =>
  Object.values(subpartMap).map(({ label, value }) => ({
    label,
    suffix: !value ? slugify(label) : value(),
  }));

/**
 * Gets an object property by path, if path is nested is executed recursively
 * @param object the source object
 * @param path the desired path
 */
export const getObjectProperty = (object: any, path: string): any => {
  if (!object) {
    return undefined;
  }

  if (!path.includes(".")) {
    return object[path];
  }

  const pathParts = path.split(".");
  const basePath = pathParts[0];
  if (!object[basePath]) {
    return undefined;
  }

  const baseObject = object[basePath];
  return getObjectProperty(baseObject, path.replace(`${basePath}.`, ""));
};

/**
 * Creates slug or kebab-case string from value
 * @param value the value to slugify
 */
export const slugify = (value: string) =>
  value
    .toLowerCase()
    .replace(/\s+/g, "-")
    .replace(/[^\w-]+/g, "");

/**
 * Suppresses default behaviour and propagation of a given event
 * @param event the event to suppress
 */
export const suppressEvent = (event: ChangeEvent<any> | MouseEvent<any, MouseEvent> | null) => {
  if (!event) {
    return;
  }

  event.preventDefault();
  event.stopPropagation();
};
