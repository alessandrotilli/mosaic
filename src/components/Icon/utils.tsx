import React from "react";
import MUIAccountCircleIcon from "@material-ui/icons/AccountCircle";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIAppsIcon from "@material-ui/icons/Apps";
import MUIArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import MUIBlockIcon from "@material-ui/icons/Block";
import MUIBrightness4Icon from "@material-ui/icons/Brightness4";
import MUIBusinessIcon from "@material-ui/icons/Business";
import MUICheckIcon from "@material-ui/icons/Check";
import MUICheckBoxIcon from "@material-ui/icons/CheckBox";
import MUICheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import MUIChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import MUIChevronRightIcon from "@material-ui/icons/ChevronRight";
import MUICloseIcon from "@material-ui/icons/Close";
import MUICropIcon from "@material-ui/icons/Crop";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIDetailsIcon from "@material-ui/icons/Details";
import MUIDraftsIcon from "@material-ui/icons/Drafts";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIErrorIcon from "@material-ui/icons/Error";
import MUIFilterListIcon from "@material-ui/icons/FilterList";
import MUIFirstPageIcon from "@material-ui/icons/FirstPage";
import MUIGroupWorkIcon from "@material-ui/icons/GroupWork";
import MUIHomeIcon from "@material-ui/icons/Home";
import MUIKeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import MUIKeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
import MUILanguageIcon from "@material-ui/icons/Language";
import MUILastPageIcon from "@material-ui/icons/LastPage";
import MUIListIcon from "@material-ui/icons/List";
import MUIMailIcon from "@material-ui/icons/Mail";
import MUIMenuIcon from "@material-ui/icons/Menu";
import MUINotificationsIcon from "@material-ui/icons/Notifications";
import MUIOpenInNew from "@material-ui/icons/OpenInNew";
import MUIPauseIcon from "@material-ui/icons/Pause";
import MUIPlayArrowIcon from "@material-ui/icons/PlayArrow";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUIPowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import MUIPrintIcon from "@material-ui/icons/Print";
import MUIRefreshIcon from "@material-ui/icons/Refresh";
import MUISaveIcon from "@material-ui/icons/Save";
import MUIScheduleIcon from "@material-ui/icons/Schedule";
import MUISearchIcon from "@material-ui/icons/Search";
import MUISendIcon from "@material-ui/icons/Send";
import MUISettingsIcon from "@material-ui/icons/Settings";
import MUISignalWifi1BarOutlinedIcon from "@material-ui/icons/SignalWifi1BarOutlined";
import MUITimelineIcon from "@material-ui/icons/Timeline";
import MUIVisibilityIcon from "@material-ui/icons/Visibility";
import MUIVisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import MUIWarningIcon from "@material-ui/icons/Warning";
import MUIWifiOffOutlinedIcon from "@material-ui/icons/WifiOffOutlined";
import MUIWifiOutlinedIcon from "@material-ui/icons/WifiOutlined";

import { Icons, IIconsCatalog } from "../../types/Icon";

export const iconsCatalog: IIconsCatalog = {
  [Icons.account]: <MUIAccountCircleIcon />,
  [Icons.add]: <MUIAddIcon />,
  [Icons.apps]: <MUIAppsIcon />,
  [Icons.block]: <MUIBlockIcon />,
  [Icons.brightness]: <MUIBrightness4Icon />,
  [Icons.business]: <MUIBusinessIcon />,
  [Icons.chart_timeline]: <MUITimelineIcon />,
  [Icons.check]: <MUICheckIcon />,
  [Icons.checkbox]: <MUICheckBoxIcon />,
  [Icons.checkbox_empty]: <MUICheckBoxOutlineBlankIcon />,
  [Icons.clock]: <MUIScheduleIcon />,
  [Icons.close]: <MUICloseIcon />,
  [Icons.connection_slow]: <MUISignalWifi1BarOutlinedIcon />,
  [Icons.crop]: <MUICropIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.details]: <MUIDetailsIcon />,
  [Icons.down]: <MUIKeyboardArrowDownIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.error]: <MUIErrorIcon />,
  [Icons.filter]: <MUIFilterListIcon />,
  [Icons.first]: <MUIFirstPageIcon />,
  [Icons.group]: <MUIGroupWorkIcon />,
  [Icons.home]: <MUIHomeIcon />,
  [Icons.language]: <MUILanguageIcon />,
  [Icons.last]: <MUILastPageIcon />,
  [Icons.left]: <MUIChevronLeftIcon />,
  [Icons.list_check]: <MUIPlaylistAddCheckIcon />,
  [Icons.list]: <MUIListIcon />,
  [Icons.mail_open]: <MUIDraftsIcon />,
  [Icons.mail]: <MUIMailIcon />,
  [Icons.menu]: <MUIMenuIcon />,
  [Icons.notifications]: <MUINotificationsIcon />,
  [Icons.open_new]: <MUIOpenInNew />,
  [Icons.pause]: <MUIPauseIcon />,
  [Icons.play]: <MUIPlayArrowIcon />,
  [Icons.power]: <MUIPowerSettingsNewIcon />,
  [Icons.print]: <MUIPrintIcon />,
  [Icons.refresh]: <MUIRefreshIcon />,
  [Icons.right]: <MUIChevronRightIcon />,
  [Icons.save]: <MUISaveIcon />,
  [Icons.search]: <MUISearchIcon />,
  [Icons.send]: <MUISendIcon />,
  [Icons.settings]: <MUISettingsIcon />,
  [Icons.up_arrow]: <MUIArrowUpwardIcon />,
  [Icons.up]: <MUIKeyboardArrowUpIcon />,
  [Icons.visibility]: <MUIVisibilityIcon />,
  [Icons.visibility_off]: <MUIVisibilityOffIcon />,
  [Icons.warn]: <MUIWarningIcon />,
  [Icons.wifi_off]: <MUIWifiOffOutlinedIcon />,
  [Icons.wifi_on]: <MUIWifiOutlinedIcon />,
};
