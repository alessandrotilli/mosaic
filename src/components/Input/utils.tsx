import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
