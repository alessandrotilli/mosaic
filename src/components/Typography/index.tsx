import React, { FC, useMemo } from "react";
import {
  Skeleton as MUISkeleton,
  Typography as MUITypography,
  TypographyVariant as MUITypographyVariant,
} from "@mui/material";

import { ITypography, TypographyVariants } from "../../types/Typography";
import localized, { ILocalizableProperty } from "../../utils/hocs/localized";

const VARIANT_COMPONENT_MAP: { [key in MUITypographyVariant]?: string } = {
  body1: "p",
  caption: "span",
  h5: "h1",
  h6: "h2",
  overline: "h4",
  subtitle1: "h3",
};

export const DATA_CY_DEFAULT = "typography";
export const LOCALIZABLE_PROPS: ILocalizableProperty[] = [
  { name: "children", type: "string" },
  { name: "content", type: "string" },
];

const Typography: FC<ITypography> = ({
  bottomSpacing = false,
  children,
  content,
  dataCy = DATA_CY_DEFAULT,
  display = "initial",
  loading = false,
  style,
  truncated = false,
  variant = "body",
}) => {
  const muiVariant = useMemo((): MUITypographyVariant => {
    switch (variant) {
      case "body":
      case TypographyVariants.body:
        return "body1";
      case "caption":
      case TypographyVariants.caption:
        return "caption";
      case "label":
      case TypographyVariants.label:
        return "subtitle1";
      case "overline":
      case TypographyVariants.overline:
        return "overline";
      case "pagetitle":
      case TypographyVariants.pagetitle:
        return "h5";
      case "title":
      case TypographyVariants.title:
        return "h6";
      default:
        return "body1";
    }
  }, [variant]);

  return (
    <MUITypography
      data-cy={`${dataCy}${loading ? "-loading" : ""}`}
      display={display}
      gutterBottom={bottomSpacing}
      noWrap={truncated}
      style={style}
      variant={muiVariant}
      variantMapping={VARIANT_COMPONENT_MAP}
    >
      {loading ? <MUISkeleton /> : content || children}
    </MUITypography>
  );
};

export const TypographyWithProps = Typography;

export default localized(Typography, {
  dataCyShortcut: "dataCy",
  localizableProps: LOCALIZABLE_PROPS,
});
