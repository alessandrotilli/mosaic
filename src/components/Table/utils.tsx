import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionPosition, ITableAction, ITableColumn } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";
import { getDataCyForSubComponent, suppressEvent } from "../../utils";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: ITableAction): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    position = TableActionPosition.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    onClick: (event, data) => {
      if (event) {
        suppressEvent(event);
      }
      callback(data);
    },
    position,
  };
};

export const actionComponentAdapter = (props: any, dataCy: string) => {
  const { data, action } = props;
  const { disabled, hidden, icon, position, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const displaysButton = position === TableActionPosition.default;
  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {displaysButton && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={() => onClick(undefined, data)}
          />
        </Fragment>
      )}
      {!displaysButton && (
        <Fragment>
          <IconButton
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={iconName}
            onClick={() => onClick(undefined, data)}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: ITableColumn): MTColumnType<object> => {
  const { label, path, render, width } = column;
  return {
    ...(!!width && ({ width } as object)),
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
