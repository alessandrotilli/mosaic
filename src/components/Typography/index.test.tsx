import renderer from "react-test-renderer";
import { ITypography, TypographyVariants, TypographyDisplay } from "../../types/Typography";
import { getLocalizedTestable } from "../../utils/tests";
import Typography, { DATA_CY_DEFAULT } from ".";
import { MessageMock, mockedMessages, LocaleMock } from "../../utils/mocks/IntlProviderMock";

const defaultProps: ITypography = {
  content: "Text",
};

const getTypographyTestable = (props?: ITypography, dataCy = DATA_CY_DEFAULT, domNode = "p") =>
  getLocalizedTestable(Typography, { dataCy, domNode, props: { ...defaultProps, ...props } });

describe("Typography test suite:", () => {
  it("default", () => {
    const { element, wrapper } = getTypographyTestable();
    expect(wrapper).toHaveLength(1);
    expect(wrapper.hasClass("MuiTypography-body1"));

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("dataCy", () => {
    const { element, wrapper } = getTypographyTestable({ dataCy: "custom" }, "custom");
    expect(wrapper).toHaveLength(1);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("localized", () => {
    const { element, wrapper } = getTypographyTestable({
      ...defaultProps,
      content: MessageMock.typography,
      localized: true,
    });
    expect(wrapper.text()).toEqual(mockedMessages[LocaleMock.en][MessageMock.typography]);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("bottomSpacing", () => {
    const { element, wrapper } = getTypographyTestable({ bottomSpacing: true });
    expect(wrapper.hasClass("MuiTypography-gutterBottom"));

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("display", () => {
    const { element, wrapper } = getTypographyTestable({ display: TypographyDisplay.inline });
    expect(wrapper.hasClass("MuiTypography-displayInline"));

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("loading", () => {
    const { element, wrapper } = getTypographyTestable({ loading: true }, `${DATA_CY_DEFAULT}-loading`);
    const placeholder = wrapper.find("span.MuiSkeleton-root");
    expect(placeholder).toHaveLength(1);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("truncated", () => {
    const { element, wrapper } = getTypographyTestable({ truncated: true });
    expect(wrapper.hasClass("MuiTypography-noWrap"));

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("variant", () => {
    const { element, wrapper } = getTypographyTestable({ variant: TypographyVariants.title }, undefined, "h2");
    expect(wrapper.hasClass("MuiTypography-h6"));

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });
});
