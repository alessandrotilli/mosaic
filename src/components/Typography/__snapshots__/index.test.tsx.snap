// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography test suite: bottomSpacing 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
  data-cy="typography"
>
  Text
</p>
`;

exports[`Typography test suite: dataCy 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
  data-cy="custom"
>
  Text
</p>
`;

exports[`Typography test suite: default 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
  data-cy="typography"
>
  Text
</p>
`;

exports[`Typography test suite: display 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 MuiTypography-displayInline"
  data-cy="typography"
>
  Text
</p>
`;

exports[`Typography test suite: loading 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
  data-cy="typography-loading"
>
  <span
    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
</p>
`;

exports[`Typography test suite: localized 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
  data-cy="typography"
>
  Typography
</p>
`;

exports[`Typography test suite: truncated 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
  data-cy="typography"
>
  Text
</p>
`;

exports[`Typography test suite: variant 1`] = `
<h2
  className="MuiTypography-root MuiTypography-h6"
  data-cy="typography"
>
  Text
</h2>
`;
